Source/ObjectiveDropboxOfficial/Shared/Handwritten/OAuth/DBOAuthManager.m

```diff
@@ -171,20 +187,30 @@ - (NSURL *)authURL {
 
   NSString *localeIdentifier = [[NSBundle mainBundle] preferredLocalizations].firstObject ?: @"en";
 
-  // used to prevent malicious impersonation of app from web browser
-  NSString *state = [[NSProcessInfo processInfo] globallyUniqueString];
-  [[NSUserDefaults standardUserDefaults] setValue:state forKey:kDBSDKCSERFKey];
-
-  components.queryItems = @[
-    [NSURLQueryItem queryItemWithName:@"response_type" value:@"token"],
+  NSMutableArray<NSURLQueryItem *> *params = [@[
     [NSURLQueryItem queryItemWithName:@"client_id" value:_appKey],
-    [NSURLQueryItem queryItemWithName:@"redirect_uri" value:[_redirectURL absoluteString]],
-    [NSURLQueryItem queryItemWithName:@"disable_signup" value:self.disableSignup ? @"true" : @"false"],
-    [NSURLQueryItem queryItemWithName:@"force_reauthentication"
-                                value:self.webAuthShouldForceReauthentication ? @"true" : @"false"],
+    [NSURLQueryItem queryItemWithName:@"redirect_uri" value:_redirectURL.absoluteString],
+    [NSURLQueryItem queryItemWithName:@"disable_signup" value:_disableSignup ? @"true" : @"false"],
     [NSURLQueryItem queryItemWithName:@"locale" value:[self.locale localeIdentifier] ?: localeIdentifier],
-    [NSURLQueryItem queryItemWithName:@"state" value:state],
-  ];
+  ] mutableCopy];
+
+  NSString *state = nil;
+  if (_authSession) {
+    // Code flow
+    state = _authSession.state;
+    [params addObjectsFromArray:[DBOAuthUtils createPkceCodeFlowParamsForAuthSession:_authSession]];
+  } else {
+    // Token flow
+    state = [[NSProcessInfo processInfo] globallyUniqueString];
+    [params addObject:[NSURLQueryItem queryItemWithName:@"response_type" value:@"token"]];
+    [params addObject:[NSURLQueryItem queryItemWithName:@"state" value:state]];
+  }
+  // used to prevent malicious impersonation of app from web browser
+  [[NSUserDefaults standardUserDefaults] setValue:state forKey:kDBSDKCSRFKey];
+
+  [params addObject:[NSURLQueryItem queryItemWithName:@"force_reauthentication"
+                                                value:_webAuthShouldForceReauthentication ? @"true" : @"false"]];
+  components.queryItems = params;
   return components.URL;
```

Comment:
this url can also be nil, should we assert here? SFSafariViewController expects a non nil url. The higher up the stack the error, the better info we have.